typescript/cli/src/submitters/EV5FileSubmitter.ts

```diff
@@ -0,0 +1,36 @@
+import { Logger } from 'pino';
+
+import {
+  ProtocolTypedTransaction,
+  TxSubmitterInterface,
+  TxSubmitterType,
+} from '@hyperlane-xyz/sdk';
+import { Annotated, ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import { appendYamlOrJson } from '../utils/files.js';
+
+import { CustomTxSubmitterType, EV5FileTxSubmitterProps } from './types.js';
+
+export class EV5FileSubmitter
+  implements TxSubmitterInterface<ProtocolType.Ethereum>
+{
+  txSubmitterType: TxSubmitterType =
+    CustomTxSubmitterType.FILE as TxSubmitterType;
```

Comment:
todo: try to fix `as`

Reply:
I don't know if it will work but maybe what you can do is update the `TxSubmitterInterface` defintion to:
```typescript
export interface TxSubmitterInterface<TProtocol extends ProtocolType, TSubmitterType extends TxSubmitterType = TxSubmitterType> { ...
```

and then provide the extended enum as second generic type in the `EV5FileSubmitter` def

Reply:
i dont think its possible to extend the enum in ts.

i've updated to extend string, but default to `TxSubmitterType`